libsql-server/src/namespace/mod.rs

```diff
@@ -152,18 +157,44 @@ impl MakeNamespace for PrimaryNamespaceMaker {
         &self,
         name: NamespaceName,
         restore_option: RestoreOption,
+        bottomless_db_id: Option<String>,
         allow_creation: bool,
         _reset: ResetCb,
     ) -> crate::Result<Namespace<Self::Database>> {
-        Namespace::new_primary(&self.config, name, restore_option, allow_creation).await
+        Namespace::new_primary(
+            &self.config,
+            name,
+            restore_option,
+            bottomless_db_id,
+            allow_creation,
+        )
+        .await
     }
 
-    async fn destroy(&self, namespace: NamespaceName, prune_all: bool) -> crate::Result<()> {
+    async fn destroy(
+        &self,
+        namespace: NamespaceName,
+        bottomless_db_id: Option<Option<String>>,
+        prune_all: bool,
+    ) -> crate::Result<()> {
         let ns_path = self.config.base_path.join("dbs").join(namespace.as_str());
 
         if prune_all {
             if let Some(ref options) = self.config.bottomless_replication {
-                let options = make_bottomless_options(options, namespace);
+                let bottomless_db_id = match bottomless_db_id {
+                    Some(db_id) => db_id,
+                    None => {
+                        if !ns_path.try_exists()? {
+                            None
+                        } else {
+                            let db_config_store = DatabaseConfigStore::load(&ns_path)
+                                .context("Could not load database config")?;
```

Comment:
We want to avoid using context for now, as its designed to be used for cli apps and doesn't work well for our server reporting methods. The reason being that this only does the full chain print when its printed via debug not display but we use display. This will hide the internal error message and make it really hard to debug. I would instead defer to error logging at the site here the error like `tracing::error!("could not load database: {?}", err)` and then returning the error not wrapped in this context call.

Reply:
I was following the existing code but happy to improve. Done.